src/api/users.ts

```diff
@@ -1,28 +1,51 @@
 import express from "express";
 import { authUser } from "../auth.js";
 import sql from "../database.js";
+import createHttpError from "http-errors";
+import { z } from "zod";
+import { Courses } from "../schema.js";
 
 const router = express.Router();
 export default router;
 
-router.get("/users/history", (req, res) => {
-  res.send({
-    completedCourses: [12, 52, 236],
-  });
+router.get("/users/history", async (req, res) => {
+  const user = await authUser(req);
+
+  const userCourses = await sql<Pick<Courses, "id" | "prefix" | "number">[]>`
+    SELECT course_id AS id, prefix, number FROM user_course_history
+    JOIN courses ON courses.id = course_id 
+    WHERE user_id = ${user.id}
+  `;
+
+  res.send(userCourses.map((course) => ({ courseId: course.id, prefix: course.prefix, number: course.number })));
 });
 
-router.put("/users/history/:courseId", async (req, res) => {
+const userHistoryReqBody = z.array(z.number());
+router.put("/users/history", async (req, res) => {
   const user = await authUser(req);
-  const courseId = req.params.courseId;
+  const courseIds = userHistoryReqBody.parse(req.body);
+
+  await sql`
+    INSERT INTO user_course_history (user_id, course_id)
+    SELECT ${user.id}, course_id FROM UNNEST(${courseIds}::int[]) AS course_id
+    ON CONFLICT DO NOTHING
+  `;
```

Comment:
We should have graceful error handling here, if the user tries to input a course ID that doesn't exist (currently will kick back a 500 because it violated the FK relation). Can be done in a later pr if you want just make an issue for it.

Reply:
https://github.com/DegreePrerequisiteTracker/Degree-Prerequisite-Tracker/issues/101